### PR TITLE
Fix sorting with joins

### PR DESCRIPTION
**Pull request by borsch**
Fix for https://github.com/perplexhub/rsql-jpa-specification/issues/63

### CODE REVIEW

File: rsql-jpa/src/main/java/io/github/perplexhub/rsql/SortUtils.java [hunk 1]
```diff
@@ -31,26 +29,20 @@ static List<Order> parseSort(@Nullable final String sort, final Map<String, Stri
...
     private static Order sortToJpaOrder(final String[] parts, final Map<String, String> propertyMapper, final Root<?> root, final CriteriaBuilder cb) {
         final String property = parts[0];
         final String direction = parts[1];
 
-        final String propertyPath = propertyMapper.getOrDefault(property, property);
-        Expression<?> propertyExpression = pathToExpression(root, propertyPath);
+        final RSQLJPAPredicateConverter rsqljpaPredicateConverter = new RSQLJPAPredicateConverter(cb, propertyMapper);
+        final RSQLJPAContext rsqljpaContext = rsqljpaPredicateConverter.findPropertyPath(property, root);
+
+        Expression<?> propertyExpression = rsqljpaContext.getPath();
```
Comment (alancruzcgi) [inline rsql-jpa/src/main/java/io/github/perplexhub/rsql/SortUtils.java:40]: Interesting.

Comment (borsch) [reply rsql-jpa/src/main/java/io/github/perplexhub/rsql/SortUtils.java:40]: In this case `RSQLJPAPredicateConverter` has already built-in functionality to build correct `Path` for given `propertyPath`. Also there is already correct support for `join`. That's why it make sense to reuse it

**Comment by borsch** [comment]
@alancruzcgi I've update test. Now it's trying to sort companies by role code. Path there is following

`company -> list<user> -> list<userRole> -> role.code`

Please note that in this case there are issue with duplicates. I couldn't solve this with simple `distinct(true)` as it gives `role.code should be in group list bla-bla`. You can read here about this spring-data-jpa issue which is not solved yet https://github.com/spring-projects/spring-data-jpa/issues/1115

**UPD:** as for now I assume that you sample `parent -> list<children> -> list<grandChildren> -> list<greatGrandChildren> -> targetValue` doesn't have much different from test that I added

**Review by alancruzcgi — APPROVED** [review]
I agree, thanks for your input, info and hard work on this one @borsch. Wishing you a nice week ahead, be safe.

**Pull request merged**